examples/components/MNIST_VERTICAL/traininsilo/host.py

```diff
@@ -0,0 +1,446 @@
+"""Script for mock components."""
+import argparse
+import logging
+import sys
+import os
+from aml_comm import AMLCommSocket
+
+import mlflow
+import torch
+import pandas as pd
+from torch import nn
+from torch.optim import SGD
+from torch.utils.data import Dataset
+from torch.utils.data.dataloader import DataLoader
+
+
+class TopModel(nn.Module):
+    def __init__(self) -> None:
+        super(TopModel, self).__init__()
+        self._model = nn.Linear(512, 10)
+
+    def forward(self, x) -> torch.tensor:
+        return self._model(x.squeeze())
+
+
+class TopDataset(Dataset):
+    """Image dataset."""
+
+    def __init__(self, csv_file):
+        """
+        Args:
+            csv_file (string): Path to the csv file with annotations.
+            root_dir (string): Directory with all the images.
+            transform (callable, optional): Optional transform to be applied
+                on a sample.
+        """
+        super(TopDataset, self).__init__()
+        self.labels = pd.read_csv(csv_file, index_col=0)
+
+    def __len__(self):
+        return len(self.labels)
+
+    def __getitem__(self, idx):
+        if torch.is_tensor(idx):
+            idx = idx.tolist()
+
+        return self.labels["label"][idx]
+
+
+class MnistTrainer:
+    def __init__(
+        self,
+        global_size,
+        global_rank,
+        global_comm,
+        train_data_dir="./",
+        test_data_dir="./",
+        model_path=None,
+        lr=0.01,
+        epochs=1,
+        batch_size=64,
+        experiment_name="default-experiment",
+        iteration_num=1,
+    ):
+        """MNIST Trainer trains RESNET18 model on the MNIST dataset.
+
+        Args:
+            workers_num (int): Number of processes involved in VFL
```

Comment:
here, need to align docstring with list of args (workers_num + rank doesn't exist, also add global_*)